modules/apps/web-experience/journal/journal-service/src/main/java/com/liferay/journal/internal/upgrade/v0_0_7/UpgradeJournalArticleDates.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.journal.internal.upgrade.v0_0_7;
+
+import com.liferay.portal.kernel.dao.jdbc.AutoBatchPreparedStatementUtil;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.StringBundler;
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Timestamp;
+
+/**
+ * @author Mariano Alvaro Saiz
+ */
+public class UpgradeJournalArticleDates extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		_updateCreateDate();
+		_updateModifiedDate();
+	}
+
+	private void _updateCreateDate() throws SQLException {
```

Comment:
Hi guys,

What we want in this method is to set the createDate for the first version in the versions related to that article. I guess we can do this in just one update clause. What do think? @marianoalvarosaiz @jorgediaz-lr

Reply:
If you don't agree with this change at least I think that we can remove the usage of JournalArticleResource in the first query.

Reply:
I agree in removing JournalArticleResource table from SQL, but I think it doesn't worth joining select and update sqls.

Reply:
I have a proposal for you @marianoalvarosaiz and @jorgediaz-lr, we see it tomorrow

Reply:
We shouldn't do an update with subselect due to MySQL optimization shortcomings: https://dev.mysql.com/doc/refman/5.7/en/subquery-optimization.html

Reply:
:thumbsup: :heart_eyes: